card/mod/01_core/set/all/collection.rb

```diff
@@ -177,15 +177,57 @@ def set_search_params_variables! hash
   end
 
 
-  def each_nest args={}
-    Card::Content.new(card.content, card).find_chunks( Card::Chunk::Reference ).each do |chunk|
+  def each_reference_with_args args={}
+    Card::Content.new(_render_raw(args), card).find_chunks( Card::Chunk::Reference ).each do |chunk|
       yield(chunk.referee_name.to_s, nest_args(args,chunk))
     end
   end
 
-  def map_nests args={}, &block
+
+  def each_nested_chunk args={}
```

Comment:
Oh, I didn't realize we had code treating the referencees as a collection.  That's cool, and this looks like a big upgrade to that initial step.